[Bug] [dolphinscheduler-spi] JSONUtils parse LocalDateTime error

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.


### What happened

When I try to set Custom Parameters of SQL Task Node with OUT type, I get such error msg:

```
[ERROR] 2022-06-10 09:32:40.206 +0000 org.apache.dolphinscheduler.spi.utils.JSONUtils:[215] - json to map exception!
com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot deserialize instance of `java.lang.String` out of START_OBJECT token
 at [Source: (String)"{"id":1,"code":5791423019648,"name":"cat","version":1,"description":"","project_code":5791415412992,"release_state":1,"user_id":1,"global_params":"[]","flag":1,"locations":"[{\"taskCode\":5791416706432,\"x\":-5.5,\"y\":147.5}]","warning_group_id":null,"timeout":0,"tenant_id":-1,"execution_type":0,"create_time":{},"update_time":{}}"; line: 1, column: 313] (through reference chain: java.util.LinkedHashMap["create_time"])
        at com.fasterxml.jackson.databind.exc.MismatchedInputException.from(MismatchedInputException.java:59)
        at com.fasterxml.jackson.databind.DeserializationContext.reportInputMismatch(DeserializationContext.java:1445)
        at com.fasterxml.jackson.databind.DeserializationContext.handleUnexpectedToken(DeserializationContext.java:1219)
        at com.fasterxml.jackson.databind.DeserializationContext.handleUnexpectedToken(DeserializationContext.java:1129)
```

In msg, `create_time` had wrong value `{}`. Thought debug, I found sql task Convert datatime to LocalDateTime. And then, JSONUtils parse this LocalDateTime object to `{}`.

I write test case, this situation is reproduced:

```
    @Test
    public void localDateTimeToString() {
        TimeZone.setDefault(TimeZone.getTimeZone("Asia/Shanghai"));
        String time = "2022-02-22 13:38:24";
        Date date = DateUtils.stringToDate(time);
        LocalDateTime localDateTime = LocalDateTime.ofInstant(date.toInstant(), ZoneId.systemDefault());
        String json = JSONUtils.toJsonString(localDateTime);
        // console print:  {}
        System.out.println(json);
        Assert.assertEquals(json, "\"" + time + "\"");
    }
```



### What you expected to happen

JSONUtils parse LocalDateTime object to correct value.

### How to reproduce

Run the above test case.

### Anything else

_No response_

### Version

3.0.0-beta-1

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/dolphinscheduler-spi/src/main/java/org/apache/dolphinscheduler/spi/utils/JSONUtils.java
+++ b/dolphinscheduler-spi/src/main/java/org/apache/dolphinscheduler/spi/utils/JSONUtils.java
@@ -26,12 +26,18 @@ import static com.fasterxml.jackson.databind.MapperFeature.REQUIRE_SETTERS_FOR_G
 
 import java.io.IOException;
 import java.text.SimpleDateFormat;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
 import java.util.Map;
 import java.util.TimeZone;
 
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.module.SimpleModule;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
@@ -59,6 +65,10 @@ public class JSONUtils {
         logger.info("init timezone: {}",TimeZone.getDefault());
     }
 
+    private static final SimpleModule LOCAL_DATE_TIME_MODULE = new SimpleModule()
+            .addSerializer(LocalDateTime.class, new LocalDateTimeSerializer())
+            .addDeserializer(LocalDateTime.class, new LocalDateTimeDeserializer());
+
     /**
      * can use static singleton, inject: just make sure to reuse!
      */
@@ -67,6 +77,7 @@ public class JSONUtils {
             .configure(ACCEPT_EMPTY_ARRAY_AS_NULL_OBJECT, true)
             .configure(READ_UNKNOWN_ENUM_VALUES_AS_NULL, true)
             .configure(REQUIRE_SETTERS_FOR_GETTERS, true)
+            .registerModule(LOCAL_DATE_TIME_MODULE)
             .setTimeZone(TimeZone.getDefault())
             .setDateFormat(new SimpleDateFormat(Constants.YYYY_MM_DD_HH_MM_SS));
 
@@ -289,4 +300,27 @@ public class JSONUtils {
         }
 
     }
+
+    public static class LocalDateTimeSerializer extends JsonSerializer<LocalDateTime> {
+
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern(Constants.YYYY_MM_DD_HH_MM_SS);
+
+        @Override
+        public void serialize(LocalDateTime value,
+                              JsonGenerator gen,
+                              SerializerProvider serializers) throws IOException {
+            gen.writeString(value.format(formatter));
+        }
+    }
+
+    public static class LocalDateTimeDeserializer extends JsonDeserializer<LocalDateTime> {
+
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern(Constants.YYYY_MM_DD_HH_MM_SS);
+
+        @Override
+        public LocalDateTime deserialize(JsonParser p, DeserializationContext context) throws IOException {
+            return LocalDateTime.parse(p.getValueAsString(), formatter);
+        }
+    }
+
 }

--- a/dolphinscheduler-spi/src/test/java/org/apache/dolphinscheduler/spi/utils/JSONUtilsTest.java
+++ b/dolphinscheduler-spi/src/test/java/org/apache/dolphinscheduler/spi/utils/JSONUtilsTest.java
@@ -1,0 +1,250 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.dolphinscheduler.spi.utils;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.sql.Time;
+import java.sql.Timestamp;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TimeZone;
+
+public class JSONUtilsTest {
+
+
+    @Test
+    public void createObjectNodeTest() {
+        String jsonStr = "{\"a\":\"b\",\"b\":\"d\"}";
+
+        ObjectNode objectNode = JSONUtils.createObjectNode();
+        objectNode.put("a", "b");
+        objectNode.put("b", "d");
+        String s = JSONUtils.toJsonString(objectNode);
+        Assert.assertEquals(s, jsonStr);
+    }
+
+    @Test
+    public void toMap() {
+
+        String jsonStr = "{\"id\":\"1001\",\"name\":\"Jobs\"}";
+
+        Map<String, String> models = JSONUtils.toMap(jsonStr);
+        Assert.assertEquals("1001", models.get("id"));
+        Assert.assertEquals("Jobs", models.get("name"));
+
+    }
+
+
+    @Test
+    public void string2MapTest() {
+        String str = list2String();
+
+        List<LinkedHashMap> maps = JSONUtils.toList(str,
+                LinkedHashMap.class);
+
+        Assert.assertEquals(1, maps.size());
+        Assert.assertEquals("mysql200", maps.get(0).get("mysql service name"));
+        Assert.assertEquals("192.168.xx.xx", maps.get(0).get("mysql address"));
+        Assert.assertEquals("3306", maps.get(0).get("port"));
+        Assert.assertEquals("80", maps.get(0).get("no index of number"));
+        Assert.assertEquals("190", maps.get(0).get("database client connections"));
+    }
+
+    public String list2String() {
+
+        LinkedHashMap<String, String> map1 = new LinkedHashMap<>();
+        map1.put("mysql service name", "mysql200");
+        map1.put("mysql address", "192.168.xx.xx");
+        map1.put("port", "3306");
+        map1.put("no index of number", "80");
+        map1.put("database client connections", "190");
+
+        List<LinkedHashMap<String, String>> maps = new ArrayList<>();
+        maps.add(0, map1);
+        String resultJson = JSONUtils.toJsonString(maps);
+        return resultJson;
+    }
+
+    @Test
+    public void testParseObject() {
+        Assert.assertNull(JSONUtils.parseObject(""));
+        Assert.assertNull(JSONUtils.parseObject("foo", String.class));
+    }
+
+
+    @Test
+    public void testJsonByteArray() {
+        String str = "foo";
+        byte[] serializeByte = JSONUtils.toJsonByteArray(str);
+        String deserialize = JSONUtils.parseObject(serializeByte, String.class);
+        Assert.assertEquals(str, deserialize);
+        str = null;
+        serializeByte = JSONUtils.toJsonByteArray(str);
+        deserialize = JSONUtils.parseObject(serializeByte, String.class);
+        Assert.assertNull(deserialize);
+    }
+
+    @Test
+    public void testToList() {
+        Assert.assertEquals(new ArrayList(),
+                JSONUtils.toList("A1B2C3", null));
+        Assert.assertEquals(new ArrayList(),
+                JSONUtils.toList("", null));
+    }
+
+    @Test
+    public void testToMap() {
+        Map<String, String> map = new HashMap<>();
+        map.put("foo", "bar");
+
+        Assert.assertTrue(map.equals(JSONUtils.toMap(
+                "{\n" + "\"foo\": \"bar\"\n" + "}")));
+
+        Assert.assertFalse(map.equals(JSONUtils.toMap(
+                "{\n" + "\"bar\": \"foo\"\n" + "}")));
+
+        Assert.assertNull(JSONUtils.toMap("3"));
+        Assert.assertNull(JSONUtils.toMap(null));
+
+        String str = "{\"resourceList\":[],\"localParams\":[],\"rawScript\":\"#!/bin/bash\\necho \\\"shell-1\\\"\"}";
+        Map<String, String> m = JSONUtils.toMap(str);
+        Assert.assertNotNull(m);
+    }
+
+    @Test
+    public void testToJsonString() {
+        Map<String, Object> map = new HashMap<>();
+        map.put("foo", "bar");
+
+        Assert.assertEquals("{\"foo\":\"bar\"}",
+                JSONUtils.toJsonString(map));
+        Assert.assertEquals(String.valueOf((Object) null),
+                JSONUtils.toJsonString(null));
+
+        Assert.assertEquals("{\"foo\":\"bar\"}",
+                JSONUtils.toJsonString(map, SerializationFeature.WRITE_NULL_MAP_VALUES));
+    }
+
+    @Test
+    public void parseObject() {
+        String str = "{\"color\":\"yellow\",\"type\":\"renault\"}";
+        ObjectNode node = JSONUtils.parseObject(str);
+
+        Assert.assertEquals("yellow", node.path("color").asText());
+
+        node.put("price", 100);
+        Assert.assertEquals(100, node.path("price").asInt());
+
+        node.put("color", "red");
+        Assert.assertEquals("red", node.path("color").asText());
+    }
+
+    @Test
+    public void parseArray() {
+        String str = "[{\"color\":\"yellow\",\"type\":\"renault\"}]";
+        ArrayNode node = JSONUtils.parseArray(str);
+
+        Assert.assertEquals("yellow", node.path(0).path("color").asText());
+    }
+
+    @Test
+    public void dateToString() {
+        TimeZone.setDefault(TimeZone.getTimeZone("Asia/Shanghai"));
+        String time = "2022-02-22 13:38:24";
+        Date date = DateUtils.stringToDate(time);
+        String json = JSONUtils.toJsonString(date);
+        Assert.assertEquals(json, "\"" + time + "\"");
+
+        String errorFormatTime = "Tue Feb 22 03:50:00 UTC 2022";
+        Assert.assertNull(DateUtils.stringToDate(errorFormatTime));
+    }
+
+    @Test
+    public void stringToDate() {
+        TimeZone.setDefault(TimeZone.getTimeZone("Asia/Shanghai"));
+        String json = "\"2022-02-22 13:38:24\"";
+        Date date = JSONUtils.parseObject(json, Date.class);
+        Assert.assertEquals(date, DateUtils.stringToDate("2022-02-22 13:38:24"));
+    }
+
+
+    @Test
+    public void calendarToString() {
+        TimeZone.setDefault(TimeZone.getTimeZone("Asia/Shanghai"));
+        String time = "2022-02-22 13:38:24";
+        Date date = DateUtils.stringToDate(time);
+        Calendar instance = Calendar.getInstance();
+        instance.setTime(date);
+        String json = JSONUtils.toJsonString(instance);
+        Assert.assertEquals("\"" + time + "\"", json);
+    }
+
+
+    @Test
+    public void sqlTimeToString() {
+        TimeZone.setDefault(TimeZone.getTimeZone("Asia/Shanghai"));
+        Time time = Time.valueOf("13:38:24");
+        String json = JSONUtils.toJsonString(time);
+        Assert.assertEquals("\"" + time + "\"", json);
+    }
+
+    @Test
+    public void sqlTimestampToString() {
+        TimeZone.setDefault(TimeZone.getTimeZone("Asia/Shanghai"));
+        String time = "2022-02-22 13:38:24";
+        Timestamp timestamp = Timestamp.valueOf(time);
+        String json = JSONUtils.toJsonString(timestamp);
+        Assert.assertEquals("\"" + time + "\"", json);
+    }
+
+    @Test
+    public void localDateTimeToString() {
+        TimeZone.setDefault(TimeZone.getTimeZone("Asia/Shanghai"));
+        String time = "2022-02-22 13:38:24";
+        Date date = DateUtils.stringToDate(time);
+        LocalDateTime localDateTime = LocalDateTime.ofInstant(date.toInstant(), ZoneId.systemDefault());
+        String json = JSONUtils.toJsonString(localDateTime);
+        Assert.assertEquals("\"" + time + "\"", json);
+    }
+
+    @Test
+    public void stringToLocalDateTime() {
+        TimeZone.setDefault(TimeZone.getTimeZone("Asia/Shanghai"));
+        String time = "2022-02-22 13:38:24";
+        Date date = DateUtils.stringToDate(time);
+        LocalDateTime localDateTime = LocalDateTime.ofInstant(date.toInstant(), ZoneId.systemDefault());
+        List<LocalDateTime> timeList = JSONUtils.parseObject("[\"2022-02-22 13:38:24\"]", new TypeReference<List<LocalDateTime>>(){});
+        Assert.assertNotNull(timeList);
+        Assert.assertEquals(1, timeList.size());
+        Assert.assertEquals(localDateTime, timeList.get(0));
+    }
+
+}

--- a/dolphinscheduler-spi/src/test/java/org/apache/dolphinscheduler/spi/utils/JSONUtilsTest.java
+++ b/dolphinscheduler-spi/src/test/java/org/apache/dolphinscheduler/spi/utils/JSONUtilsTest.java
@@ -176,56 +176,6 @@ public class JSONUtilsTest {
     }
 
     @Test
-    public void dateToString() {
-        TimeZone.setDefault(TimeZone.getTimeZone("Asia/Shanghai"));
-        String time = "2022-02-22 13:38:24";
-        Date date = DateUtils.stringToDate(time);
-        String json = JSONUtils.toJsonString(date);
-        Assert.assertEquals(json, "\"" + time + "\"");
-
-        String errorFormatTime = "Tue Feb 22 03:50:00 UTC 2022";
-        Assert.assertNull(DateUtils.stringToDate(errorFormatTime));
-    }
-
-    @Test
-    public void stringToDate() {
-        TimeZone.setDefault(TimeZone.getTimeZone("Asia/Shanghai"));
-        String json = "\"2022-02-22 13:38:24\"";
-        Date date = JSONUtils.parseObject(json, Date.class);
-        Assert.assertEquals(date, DateUtils.stringToDate("2022-02-22 13:38:24"));
-    }
-
-
-    @Test
-    public void calendarToString() {
-        TimeZone.setDefault(TimeZone.getTimeZone("Asia/Shanghai"));
-        String time = "2022-02-22 13:38:24";
-        Date date = DateUtils.stringToDate(time);
-        Calendar instance = Calendar.getInstance();
-        instance.setTime(date);
-        String json = JSONUtils.toJsonString(instance);
-        Assert.assertEquals("\"" + time + "\"", json);
-    }
-
-
-    @Test
-    public void sqlTimeToString() {
-        TimeZone.setDefault(TimeZone.getTimeZone("Asia/Shanghai"));
-        Time time = Time.valueOf("13:38:24");
-        String json = JSONUtils.toJsonString(time);
-        Assert.assertEquals("\"" + time + "\"", json);
-    }
-
-    @Test
-    public void sqlTimestampToString() {
-        TimeZone.setDefault(TimeZone.getTimeZone("Asia/Shanghai"));
-        String time = "2022-02-22 13:38:24";
-        Timestamp timestamp = Timestamp.valueOf(time);
-        String json = JSONUtils.toJsonString(timestamp);
-        Assert.assertEquals("\"" + time + "\"", json);
-    }
-
-    @Test
     public void localDateTimeToString() {
         TimeZone.setDefault(TimeZone.getTimeZone("Asia/Shanghai"));
         String time = "2022-02-22 13:38:24";

--- a/dolphinscheduler-spi/src/test/java/org/apache/dolphinscheduler/spi/utils/JSONUtilsTest.java
+++ b/dolphinscheduler-spi/src/test/java/org/apache/dolphinscheduler/spi/utils/JSONUtilsTest.java
@@ -24,12 +24,9 @@ import com.fasterxml.jackson.databind.node.ObjectNode;
 import org.junit.Assert;
 import org.junit.Test;
 
-import java.sql.Time;
-import java.sql.Timestamp;
 import java.time.LocalDateTime;
 import java.time.ZoneId;
 import java.util.ArrayList;
-import java.util.Calendar;
 import java.util.Date;
 import java.util.HashMap;
 import java.util.LinkedHashMap;

